workshop/README.md

```diff
@@ -201,46 +277,65 @@ If you want to be absolutely sure the transaction has been accepted and included
 
     async prepare(store) {
         await store.account.cache([
-			{
-				address: this.senderId,
-			},
-		]);
+		    {
+			    address: this.senderId,
+		    },
+	    ]);
     }
 </details>
 
 <details>
     <summary>Undo() function:</summary>
 
     undoAsset(store) {
-        const sender = store.account.get(this.senderId);
-
-		// Rollback invoice count and IDs
-		sender.asset.invoiceCount = sender.asset.invoiceCount === 1 ? undefined : --sender.asset.invoiceCount;
-		sender.asset.invoicesSent = sender.asset.invoicesSent.length === 1 ? undefined : sender.asset.invoicesSent.splice(
-			sender.asset.invoicesSent.indexOf(this.id),
-			1,
-		);
-		store.account.set(sender.address, sender);
-        return [];
-    }
+		const sender = store.account.get(this.senderId);
+		const errors = [];
+		const invoiceId = sender.asset.invoicesSent.find(id => id === this.id);
+
+		if (invoiceId === undefined || sender.asset.invoiceCount === 0) {
+			errors.push(
+				new TransactionError(
+					'Invoice ID does not exist in sender.asset.invoicesSent',
+					this.id,
+					'sender.asset.invoicesSent',
+					sender.asset.invoicesSent,
+					'A string value',
+				),
+			);
+		} else {
+			// Undo logic comes here
+			sender.asset.invoiceCount--;
+			sender.asset.invoicesSent = sender.asset.invoicesSent.filter(
+				id => id !== this.id,
+			);
+
+			store.account.set(sender.address, sender); // Save updated sender account
+		}
+
+		return errors;
+	}
 </details>
+<br>
 
-## Nuggets of Knowledge for Custom Transactions
+## Nuggets of Knowledge about Custom Transactions
```

Comment:
Maybe not needed to fix now, but for the future:
This section is breaking a little the flow of the workshop.
As I see it, the knowledge nuggets we provide here are also needed in the second part of the workshop, when implementing the payment transaction.
To not break the flow, I would explain every "nugget" at the moment when it gets relevant in the workshop. At this point, the users might be confused how deep they need to dive into the topics to finish the workshop, or if this knowledge is needed in this context at all.

Reply:
Hmm yea I agree, but there are some many nuggets of knowledge that are important that I decided to add them in between. Otherwise, the user will be bombarded with reading sections. The idea is that they can implement the payment transaction on their selves uninterrupted but I see the point. Let's ask for feedback about this during the workshop to see if we need to change this format!